src/hdmf/common/resources.py

```diff
@@ -0,0 +1,116 @@
+import numpy as np
+
+from . import register_class
+from ..container import Table, Container
+
+from ..utils import docval, call_docval_func, popargs
+
+
+def _check_id(table, id):
+    if id >= 0 and len(table.which(id=id)) == 0:
+        return np.uint64(id)
+    else:
+        raise ValueError('id must be a non-negative integer that is not already in the table: %d' % id)
+
+
+@register_class('ResourceReferenceMap')
+class ResourceReferenceMap(Table):
+
```

Comment:
It would be nice to have docstrings on all classes in resources.py to describe what each of these classes is for.